lib/experimental/fonts-library/class-wp-font-family.php

```diff
@@ -0,0 +1,415 @@
+<?php
+/**
+ * Font Family class.
+ *
+ * This file contains the Font Family class definition.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts Library
+ * @since      X.X.X
+ */
+
+if ( class_exists( 'WP_Font_Family' ) ) {
+	return;
+}
+
+
+/**
+ * Fonts Library class.
+ */
+class WP_Font_Family {
+
+	/**
+	 * Font family data
+	 *
+	 * @var array
+	 */
+	private $data;
+
+
+	/**
+	 * WP_Font_Family constructor.
+	 *
+	 * @param array $font_family Font family data.
+	 */
+	public function __construct( $font_family = array() ) {
+		$this->data = $font_family;
+	}
+
+	/**
+	 * Returns the font family data.
+	 *
+	 * @return array An array in fontFamily theme.json format.
+	 */
+	public function get_data() {
+		return $this->data;
+	}
+
+	/**
+	 * Returns the font family data.
+	 *
+	 * @return string fontFamily in theme.json format as stringified JSON.
+	 */
+	public function get_data_as_json() {
+		return wp_json_encode( $this->get_data() );
+	}
+
+	/**
+	 * Returns if the font family has font faces defined.
+	 *
+	 * @return bool true if the font family has font faces defined, false otherwise.
+	 */
+	public function has_font_faces() {
+		return (
+			isset( $this->data['fontFace'] )
+			&& is_array( $this->data['fontFace'] )
+			&& ! empty( $this->data['fontFace'] )
+		);
+	}
+
+	/**
+	 * Removes font family assets.
+	 *
+	 * @return bool True if assets were removed, false otherwise.
+	 */
+	private function remove_font_family_assets() {
+		if ( $this->has_font_faces() ) {
+			foreach ( $this->data['fontFace'] as $font_face ) {
+				$were_assets_removed = $this->delete_font_face_assets( $font_face );
+				if ( false === $were_assets_removed ) {
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Removes a font family from the database and deletes its assets.
+	 *
+	 * @return bool|WP_Error True if the font family was uninstalled, WP_Error otherwise.
+	 */
+	public function uninstall() {
+		$post = $this->get_data_from_post();
+		if ( null === $post ) {
+			return new WP_Error( 'font_family_not_found', __( 'The font family could not be found.', 'gutenberg' ) );
+		}
+		$were_assets_removed = $this->remove_font_family_assets();
+		if ( true === $were_assets_removed ) {
+			$was_post_deleted = wp_delete_post( $post->ID, true );
+			if ( null === $was_post_deleted ) {
```

Comment:
A few notes here:
- This outputs an error if the font family assets could be removed, but the post couldn't be deleted. It seems there should be an error if _either_ of these fail.
- `wp_delete_post()` returns `false` _or_ `null` on failure.
- As these are single-use variables and the method/function names are self-documenting enough, I'd suggest moving the calls directly into a joint condition instead of storing in a variable.

```php
if ( ! $this->remove_font_family_assets() || ! wp_delete_post( $post->ID, true ) instanceof WP_Post ) {
	return new WP_Error( 'font_family_not_deleted', __( 'The font family could not be deleted.', 'gutenberg' ) );
}

Reply:
Great idea, done!